CHANGELOG.md

```diff
@@ -27,6 +27,7 @@ Please add to the relevant subsections under Unreleased below on every PR where
 - Fix "UnicodeDecodeError: 'charmap'" when opening HTML files on Windows by specifying UTF-8 encoding by @KilnOfTheSecondFlame in https://github.com/wandb/wandb/pull/7730
 - Ensure `Artifact.delete()` on linked artifacts only removes link, not source artifact by @tonyyli-wandb in https://github.com/wandb/wandb/pull/7742
 - Sweep runs no longer appear to be resumed when they are not by @TimH98 https://github.com/wandb/wandb/pull/7684
+- Fix OSError encountered on `Artifact.download(skip_cache=True)` when encountering different filesystems by @tonyyli-wandb in https://github.com/wandb/wandb/pull/7835
```

Comment:
Could you also please update the pr title? It sounds like its related to reference artifacts, when its not imo

Reply:
(moving slack reply to here)

> It sounds like its related to reference artifacts, when its not imo

This is a good point, thanks.  Yeah, I agree the underlying issue isn't really about reference artifacts

Really the underlying issue looks like it's that:
- during artifact.download(...), Python was creating a temporary file on `filesystem A` (for atomic writes)
- the actual cache was located on `filesystem B`

This is speculation from what I could gather from traceback/logs, but this could plausibly happen if `filesystem A` and `filesystem B` are different partitions / one or both are mounted / etc.